test/cucumber/http_request.go

```diff
@@ -187,22 +188,47 @@ func (s *TestScenario) iWaitUpToSecondsForAResponseJsonEvent(timeout float64) er
 	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(timeout*float64(time.Second)))
 	defer cancel()
 
+	var err error
 	select {
 	case event := <-session.EventStreamEvents:
-
 		session.respJson = event
-		var err error
 		session.RespBytes, err = json.Marshal(event)
-		if err != nil {
-			return err
-		}
+		return err
 	case <-ctx.Done():
+		return fmt.Errorf("a Response Json event was not received in %v, last error was %w", time.Duration(timeout*float64(time.Second)), err)
```

Comment:
Yay. I came to the same [conclusion](https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1686#discussion_r1169863947) in my `Processor` work.